File: 2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.cpp
class Solution {
public:
    bool checkAlmostEquivalent(string word1, string word2) {
        // Brute Force
        for(char ch = 'a'; ch <= 'z'; ch++){
            int freq1 = 0;
            int freq2 = 0;
            for(auto &i: word1){
                if(i == ch) freq1++;
            }
            for(auto &i: word2){
                if(i == ch) freq2++;
            }
            
            if(abs(freq1 - freq2) > 3) return false;       
        }
        return true;
    }
};

File: 2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.java
class Solution {
    public boolean checkAlmostEquivalent(String word1, String word2) {
        // Time Complexity =>  O(26 + N + M)
        int freq1[] = new int[26];
        // a - > 0
        // b - > 1
        // c - > 2
        // z - > 25
        
      
        for(int i = 0; i < word1.length(); i++){
            // to find the ascii value 
            freq1[word1.charAt(i) - 'a']++;
        }
        
        for(int i = 0; i < word2.length(); i++){
            freq1[word2.charAt(i) - 'a']--;
        }
        
        for(int i = 0; i < 26; i++){
            if(Math.abs(freq1[i]) > 3)    return false;
        }
        return true;
    }
}